### ISSUE TITLE
Pull: new XSLT object is not activated before program, and so makes program activation fail

### ISSUE THREAD

**Issue by sandraros** [abapGit/abapGit#5262]
Programs which refer statically to new transformation objects can't be activated because the transformation has not been activated.

abapGit 1.114.0
ABAP 7.52 SP 0

While pulling https://github.com/abap2xlsx/abap2xlsx which contains the new transformation object `ZEXCEL_TR_DOCPROPS_CORE`, I get this error:

First mention of the activation issue (NB: in this list of inactive objects, I don't see the new transformation object):
![image](https://user-images.githubusercontent.com/34005250/148803963-d6f5454f-a35f-4bde-93b8-258ed283e9be.png)

Details of the syntax errors:
![image](https://user-images.githubusercontent.com/34005250/148804205-94693954-129e-488c-a1cf-58bc4835c2df.png)

We see the transformation is currently inactive:
![Transaction STRANS showing ST transformation](https://user-images.githubusercontent.com/34005250/148804290-b2ccb4f2-defb-4e27-8e30-6d981b7988d1.png)

I activated the transformation:
![activated the transformation](https://user-images.githubusercontent.com/34005250/148804798-d6f43c18-0465-4d10-9d02-cc68f01133a4.png)

After that, I can continue the activation of other objects.

**Comment by larshp** [comment]
thanks, it might be a regression from https://github.com/abapGit/abapGit/pull/5198

first step, try adding another unit test in https://github.com/abapGit/abapGit/blob/main/src/objects/core/zcl_abapgit_file_deserialize.clas.testclasses.abap for XSLT and PROG

**Comment by sandraros** [comment]
For information, I have created [this repository](https://github.com/sandraros/abapgit-reproduce-issue-5262) to contain all known test cases for XSLT, to extend your [test repository](https://github.com/abapGit-tests/XSLT_ref). Feel free to do whatever you want with it.

**Comment by sandraros** [comment]
Thanks, I will work on it in 2 hours.

**Comment by mbtools** [comment]
Tested with https://github.com/sandraros/abapgit-reproduce-issue-5262. Works flawlessly:

![SAPgui_537](https://user-images.githubusercontent.com/59966492/149204867-ffc6fc2e-c155-4272-a56c-8a3c15c6f0e1.gif)

`XSLTs` are processed twice: first in step "ABAP". They are therefore activated together with `PROGs`. If an `XSLT` depends on other objects, it won't activate. Therefore, `XSLTs` are activated again in the "LATE" step.

In my test, I used the "Activate without showing popup" setting, so there are no activate popups. Without the setting, you will get two popups but it activates the `XSLTs` properly, too.

In short, the test repo is not sufficient to reproduce the issue you mentioned initially.

**Comment by sandraros** [comment]
Could you try without using the "Activate without showing popup" setting? Without this setting, you get the popup described in the issue ("Activate despite errors"). If you click "Activate", it will activate, but if you click Edit, you see the syntax errors.

**Comment by sandraros** [comment]
Maybe the difference is because I use 7.52 SP 0 ?

**Comment by mbtools** [comment]
here's the recording without the setting (7.50 SP 21):

![SAPgui_539](https://user-images.githubusercontent.com/59966492/149219065-ed3058e2-2848-4ddf-b4a1-a6b03c48ec49.gif)

I know the kind of "Activate/Edit/Cancel" popup, but I'm not getting that. I will test on 7.54 tomorrow.

**Comment by sandraros** [comment]
This is what I get in 7.52 SP 1 / kernel 753 SP 500 :
![abapgit-issue-5262](https://user-images.githubusercontent.com/34005250/149221207-408b28f7-b1ff-496d-bb50-06456a24df60.gif)

**Comment by mbtools** [comment]
Test on 7.54 SP 2 (current ABAP 1909 Dev Edition, Kernel 777 patch 200)... It works exactly the same as my 7.50 test, with or without the setting. 

Some corrections must be missing in your system. I found the SAP Note... (we should have looked earlier):

2859159 - XSLT: Support dependencies on inactive objects
https://launchpad.support.sap.com/#/notes/2859159

**Comment by sandraros** [comment]
Thanks for investigating. I guess it must be the error, and I propose to close the issue if you agree. Thanks again!

**Comment by AndreaBorgia-Abo** [comment]
For curiosity I had a look on my own demo systems as well:
* 7.52SP04 still complains (kernel 753)
* HANA1909 works fine
I've tried uploading SAR files to install the notes (2859159 and dependents), but 2859159 itself does not meet the kernel requirements: I have 400, minimum is 521, bummer!

Oh, unrelated question: what did you guys use for making the animated gif above?

**Comment by sandraros** [comment]
Trial version 7.52 SP 1 has kernel 753 SP 500. It's fixed with the note above + you need 753 SP 521, 773 SP 230 or 777 SP 36.
ScreenToGif.

**Comment by mbtools** [comment]
I'm using ShareX. Love it. Free, too

https://getsharex.com/
